src/main/js/core/services.js

```diff
@@ -17,18 +17,13 @@ app.service('GAWTUtils', [ function() {
          });
       },
       convertHexToRgb: function (hexVal) {
-         function hexToR(h) {return parseInt((cutHex(h)).substring(0,2),16);}
-         function hexToG(h) {return parseInt((cutHex(h)).substring(2,4),16);}
-         function hexToB(h) {return parseInt((cutHex(h)).substring(4,6),16);}
-         function cutHex(h) {return (h.charAt(0)==="#") ? h.substring(1,7):h;}
-         return [hexToR(hexVal),hexToG(hexVal),hexToB(hexVal)];
+          hexVal = parseInt(hexVal.charAt(0) === '#'? hexVal.substring(1,7): hexVal, 16);
+          return [Math.floor(hexVal / 65536), Math.floor((hexVal % 65536) / 256), hexVal % 256];
       },
       convertHexAndOpacityToRgbArray: function (hexVal, opacity) {
-         function hexToR(h) {return parseInt((cutHex(h)).substring(0,2),16);}
-         function hexToG(h) {return parseInt((cutHex(h)).substring(2,4),16);}
-         function hexToB(h) {return parseInt((cutHex(h)).substring(4,6),16);}
-         function cutHex(h) {return (h.charAt(0)==="#") ? h.substring(1,7):h;}
-         return [hexToR(hexVal),hexToG(hexVal),hexToB(hexVal), opacity];
+         var a = convertHexToRgb(hexVal);
```

Comment:
Hi @jaga15, thanks for the change! However, as it stands, this won't work. `convertHexToRgb` is a function on the service so calling it like you are above is like calling `window.convertHexToRgb` which is not valid, hence why the unit tests failed.

https://travis-ci.org/GeoscienceAustralia/geo-web-toolkit/builds/75538999#L712

An easy fix would be to change the service object from being created and returned inline to assigning it to a local var and using that to reference from other functions.

Also, if the function does not already have a unit test, would be awesome if you can create one so show functionality has not been altered with this change.